networkx/algorithms/euler.py

```diff
@@ -37,10 +42,18 @@ def is_eulerian(G):
     >>> nx.is_eulerian(nx.petersen_graph())
     False
 
-    Notes
-    -----
-    If the graph is not connected (or not strongly connected, for
-    directed graphs), this function returns False.
+    If you prefer to allow graphs with isolated vertices to have Eulerian circuits,
+    you can first remove such vertices and then call `is_eulerian` as below example shows.
+
+    >>> G = nx.Graph([(0, 1), (1, 2), (0, 2)])
+    >>> G.add_node(3)
+    >>> nx.is_eulerian(G)
+    False
+
+    >>> G.remove_nodes_from(list(nx.isolates(G)))
+    >>> nx.is_eulerian(G)
+    True
```

Comment:
This example does a great job of highlighting the behavior with isolated nodes. There might be a way to keep things clear but use less text, maybe something like:

```suggestion
    Graphs with isolated nodes are not Eulerian:

    >>> G = nx.Graph([(0, 1), (1, 2), (0, 2)])
    >>> G.add_node(3)
    >>> nx.is_eulerian(G)
    False
    
    Isolated nodes can be removed to evaluate whether the
    remaining graph is Eulerian:

    >>> G.remove_nodes_from(list(nx.isolates(G)))
    >>> nx.is_eulerian(G)
    True
```

This is really subjective and just a suggestion - the original wording is totally fine so feel free to ignore!